modules/contact/contact.service.js

```diff
@@ -21,21 +22,99 @@ class ContactService {
   }
 
   async addContact(data) {
-    return new Contact(data).save();
+    if (!data.mapImages.length) {
+      return new Contact(data.contact).save();
+    }
+
+    const uploadResult = await uploadFiles([
+      data.mapImages[0].image,
+      data.mapImages[1].image,
+    ]);
+    const images = await Promise.all(uploadResult).then(res => res);
+
+    if (!images) {
+      return await new Contact(data.contact).save();
+    }
+
+    return new Contact({
+      ...data.contact,
+      images: [
+        { lang: data.mapImages[0].lang, value: images[0].fileNames },
+        { lang: data.mapImages[1].lang, value: images[1].fileNames },
+      ],
+    }).save();
   }
 
-  async updateContact(id, contactData) {
-    const contact = await Contact.findByIdAndUpdate(id, contactData, {
-      new: true,
-    });
+  async updateContact(data) {
+    if (!data.mapImages.length) {
+      return await Contact.findByIdAndUpdate(data.id, data.contact, {
+        new: true,
+      });
+    }
 
-    return contact || null;
+    const contactById = await Contact.findById(data.id).lean();
+
+    if (contactById.images.length) {
+      const deletedImages = await deleteFiles([
+        ...Object.values(contactById.images[0].value),
+        ...Object.values(contactById.images[1].value),
+      ]);
+
+      if (await Promise.allSettled(deletedImages)) {
+        return await this.uploadMapImages(data);
+      }
+    } else {
+      return this.uploadMapImages(data);
+    }
+
+    return null;
   }
 
   async deleteContact(id) {
-    const contact = await Contact.findByIdAndDelete(id);
+    const contactById = await Contact.findById(id).lean();
 
-    return contact || null;
+    if (contactById.images && contactById.images.length) {
+      const deletedImages = await deleteFiles([
+        ...Object.values(contactById.images[0].value),
+        ...Object.values(contactById.images[1].value),
+      ]);
+
+      if (await Promise.allSettled(deletedImages)) {
+        return await Contact.findByIdAndDelete(id);
+      }
+    } else {
+      return await Contact.findByIdAndDelete(id);
+    }
+
+    return null;
+  }
+
+  async uploadMapImages(data) {
```

Comment:
Please, refactor this method to handle only the saving process of the images without any usage of the `Contact` model, because you already have `create` and `update` methods for this logic